src/mutation-observer.ts

```diff
@@ -51,20 +73,48 @@ const processEffects = (node: Node | TramOneElement) => {
 	}
 };
 
-// call all cleanup effects on the node
+/**
+ * call all cleanup effects on the node
+ */
 const cleanupEffects = (cleanupEffects: (() => void)[]) => {
 	cleanupEffects.forEach((cleanup) => cleanup());
 };
 
-// unobserve the reaction tied to the node, and run all cleanup effects for the node
+/**
+ * remove the association of the store with this specific element
+ */
+const removeStoreKeyAssociation = (storeKeys: string[]) => {
+	storeKeys.forEach((storeKey) => {
+		decrementKeyStoreValue(TRAM_KEY_STORE, storeKey);
+	});
+};
+
+/**
+ * remove any stores that no longer have any elements associated with them
+ * see removeStoreKeyAssociation above
+ */
+const cleanUpObservableStores = () => {
+	const observableStore = getObservableStore(TRAM_OBSERVABLE_STORE);
+	const keyStore = getKeyStore(TRAM_KEY_STORE);
+	Object.entries(keyStore).forEach(([key, observers]) => {
+		if (observers === 0) {
+			delete observableStore[key];
```

Comment:
we should also delete the key from the keystore (otherwise we'll keep trying to delete these, even when they are already gone).